code/src/java/pcgen/cdom/format/table/TableFormatManager.java

```diff
@@ -0,0 +1,191 @@
+
+	/**
+	 * The Format of any DataTable referred to by this TableFormatManager.
+	 */
+	private final FormatManager<?> resultFormat;
+
+	/**
+	 * Constructs a new TableFormatManager that will use the underlying
+	 * AbstractReferenceContext to construct and look up DataTable objects. The
+	 * DataTable should have the lookup and result formats matching the formats
+	 * of the given FormatManagers.
+	 * 
+	 * @param objDatabase
+	 *            The ObjectDatabase used to construct or look up DataTable
+	 *            objects
+	 * @param lookupFormat
+	 *            The FormatManager for the format of the Lookup column of the
+	 *            DataTable format represented by this TableFormatManager
+	 * @param resultFormat
+	 *            The FormatManager for the format of the Result column of the
+	 *            DataTable format represented by this TableFormatManager
+	 */
+	public TableFormatManager(ObjectDatabase objDatabase,
+		FormatManager<?> lookupFormat, FormatManager<?> resultFormat)
+	{
+		this.database = Objects.requireNonNull(objDatabase);
+		this.lookupFormat = Objects.requireNonNull(lookupFormat);
+		this.resultFormat = Objects.requireNonNull(resultFormat);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public DataTable convert(String inputStr)
+	{
+		//TODO Does this need validation that the lookup/result columns are appropriate?
+		return database.get(DataTable.class, inputStr);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Indirect<DataTable> convertIndirect(String inputStr)
+	{
+		/*
+		 * TODO Need validation that the lookup/result columns are appropriate?
+		 * Yes, probably during the initialization of these references, it will
+		 * need to be checked... but how? Does this need to be like Categorized
+		 * references? ugh
+		 */
+		return database.getIndirect(DataTable.class, inputStr);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String unconvert(DataTable table)
+	{
+		return table.getName();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Class<DataTable> getManagedClass()
+	{
+		return DataTable.class;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getIdentifierType()
+	{
+		StringBuilder sb = new StringBuilder();
+		sb.append("TABLE[");
+		sb.append(lookupFormat.getIdentifierType());
+		sb.append(",");
+		sb.append(resultFormat.getIdentifierType());
+		sb.append("]");
+		return sb.toString();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public FormatManager<?> getComponentManager()
+	{
+		return null;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public int hashCode()
+	{
+		return lookupFormat.hashCode() ^ resultFormat.hashCode();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean equals(Object o)
+	{
+		if (o instanceof TableFormatManager)
+		{
+			TableFormatManager other = (TableFormatManager) o;
+			return lookupFormat.equals(other.lookupFormat)
+				&& resultFormat.equals(other.resultFormat);
```

Comment:
I will change it to final, but I think the question is how much we want to worry about it... that is true almost universally across the entire java.* design, in how they originally implemented .equals()... they didn't use .getClass(), they just tested the local variables.  (The same is basically universally true across our code base.)